### ISSUE TITLE
can't get the same result using other libraries

### ISSUE THREAD

**Issue by Hossein-1991** [mathiaszinnen/focal_loss_torch#15]
Hello,

For comparison, I decided to use two different methods in calculating focal loss. One with the library that is explained here and the other with the newly published pytorch's focal loss (https://pytorch.org/vision/main/generated/torchvision.ops.sigmoid_focal_loss.html) .

Surprisingly, I did not get the same results! why is that the case?!
These are my codes:
```python
weights = torch.FloatTensor([0.25 , 0.75])
criterion = FocalLoss(gamma=2, weights=weights)

m = torch.nn.Softmax(dim=-1)
logits = torch.tensor([[-1.2352, -0.8168],
        [ 1.4019, -0.5465],
        [ 0.8415,  0.4405],
        [ 0.1093,  0.2481],
        [ 0.2785, -1.5349],
        [-0.7501,  1.3956],
        [-0.7842,  0.8283],
        [-0.9012, -1.6445],
        [-0.0615, -0.4246],
        [ 1.0127, -0.1142]])
print (torch.argmax(m(logits),1))
target = torch.tensor([0,1,1,1,0,1,0,1,1,0])
criterion(m(logits), target)
```
the output is:
```python
tensor([1, 0, 0, 1, 0, 1, 1, 0, 0, 0])
tensor(0.2568)
```
and the other one is:
```python
from torchvision.ops import sigmoid_focal_loss
p = torch.tensor([1,0,0,1,0,1,1,0,0,0], dtype = torch.float32)
target = torch.tensor([0,1,1,1,0,1,0,1,1,0], dtype = torch.float32)
sigmoid_focal_loss(p , target, gamma = 2, alpha = 0.25,reduction = 'mean')
```
and the output is:
```python
tensor(0.1497)
```

**Comment by mathiaszinnen** [comment]
Yes, thank you for fixing the issue and for pinging me @msalhab96

**Comment by mathiaszinnen** [comment]
Should be solved now

**Comment by Hossein-1991** [comment]
Thank you.
The issue solved.
